src/main/java/org/threadly/concurrent/PrioritySchedulerStatisticTracker.java

```diff
@@ -676,114 +637,66 @@ protected static void trimWindow(Deque window) {
+   * @since 4.0.0
+   */
+  protected static class StatisticWorker extends Worker {
+    private final StatsManager statsManager;
+    
+    protected StatisticWorker(WorkerPool workerPool, 
+                              ThreadFactory threadFactory, 
+                              StatsManager statsManager) {
+      super(workerPool, threadFactory);
+      
+      this.statsManager = statsManager;
+    }
+    
     @Override
-    protected void runLowPriorityTask(TaskWrapper task) throws InterruptedException {
-      Worker w = null;
-      synchronized (workersLock) {
-        if (! isShutdownFinished()) {
-          // wait for high priority tasks that have been waiting longer than us if all workers are consumed
-          long waitMs;
-          while (getCurrentPoolSize() >= getMaxPoolSize() && 
-                 availableWorkers.size() < WORKER_CONTENTION_LEVEL &&   // only care if there is worker contention
-                 ! isShutdownFinished() &&
-                 (waitMs = task.getDelayEstimateInMs() - lastHighDelayMillis) > LOW_PRIORITY_WAIT_TOLLERANCE_IN_MS) {
-            workersLock.wait(waitMs);
-            Clock.systemNanoTime(); // update for getDelayEstimateInMillis
-          }
-          
-          if (! isShutdownFinished()) {  // check again that we are still running
-            if (getCurrentPoolSize() >= getMaxPoolSize()) {
-              synchronized (statsManager.lowPriorityWorkerAvailable.getModificationLock()) {
-                statsManager.lowPriorityWorkerAvailable.add(! availableWorkers.isEmpty());
-                StatsManager.trimWindow(statsManager.lowPriorityWorkerAvailable);
-              }
-              w = getExistingWorker(Long.MAX_VALUE);
-            } else if (getCurrentPoolSize() == 0) {
-              synchronized (statsManager.lowPriorityWorkerAvailable.getModificationLock()) {
-                statsManager.lowPriorityWorkerAvailable.add(false);
-                StatsManager.trimWindow(statsManager.lowPriorityWorkerAvailable);
-              }
-              w = makeNewWorker();
-            } else {
-              w = getExistingWorker(getMaxWaitForLowPriority());
-              synchronized (statsManager.lowPriorityWorkerAvailable.getModificationLock()) {
-                statsManager.lowPriorityWorkerAvailable.add(w != null);
-                StatsManager.trimWindow(statsManager.lowPriorityWorkerAvailable);
-              }
-              if (w == null) {
-                // this means we expired past our wait time, so create a worker if we can
-                if (getCurrentPoolSize() >= getMaxPoolSize()) {
-                  // more workers were created while waiting, now have reached our max
-                  w = getExistingWorker(Long.MAX_VALUE);
-                } else {
-                  w = makeNewWorker();
-                }
-              }
-            }
-          }
+    public void nextTask(TaskWrapper task) {
+      // may not be a wrapper for internal tasks like shutdown
```

Comment:
Here is where we track how far delayed the task was...changes above needed to happen to ensure we could reliably check this here.